src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -293,7 +297,19 @@ class ShenandoahHeap : public CollectedHeap, public ShenandoahSpaceInfo {
 
   size_t _gc_no_progress_count;
 
-  // This updates the singlular, global gc state. This must happen on a safepoint.
+  double*   _gc_historical_utilization;
```

Comment:
Could we put these fields and the attendant functionality in `ShenandoahMmuTracker`?